File: CS50x/pset6/readability/readability.py
from cs50 import get_string
import string

text = get_string("Text: ")

letters = 0
words = 1
sentences = 0

# loop through string to add
for c in text:
    # if letter
    if c.isalpha():
        letters += 1
    # if space (between words)
    elif c.isspace():
        words += 1
    # if end of sentence
    elif c == "." or c == "!" or c == "?":
        sentences += 1

L = (letters / words) * 100
S = (sentences / words) * 100

# Coleman-Liau index equation
index = 0.0588 * L - 0.296 * S - 15.8

# check index to determine grade level
if index < 1:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else:
    print(f"Grade {round(index)}")

File: CS50x/pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int count_letters(string);
int count_words(string);
int count_sentences(string);

int main(void)
{
    string text = get_string("Text: ");

    int numLetters = count_letters(text);
    int numWords = count_words(text);
    int numSentences = count_sentences(text);

    float L = ((float) numLetters / numWords) * 100.0;
    float S = ((float) numSentences / numWords) * 100.0;

    float index = 0.0588 * L - 0.296 * S - 15.8;

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", (int) round(index));
    }

}

int count_letters(string s)
{
    int letters = 0;

    for (int i = 0, n = strlen(s); i < n; i++)
    {
        if (isalpha(s[i]))
        {
            letters++;
        }
    }

    return letters;
}

int count_words(string s)
{
    int words = 1;

    for (int i = 0, n = strlen(s); i < n; i++)
    {
        if (isspace(s[i]))
        {
            words++;
        }
    }

    return words;
}

int count_sentences(string s)
{
    int sentences = 0;

    for (int i = 0, n = strlen(s); i < n; i++)
    {
        if (s[i] == '.' || s[i] == '!' || s[i] == '?')
        {
            sentences++;
        }
    }

    return sentences;
}